File: Exercicios/quase_primo.py
# import datetime

n,k = [int(x) for x in input().split()]
nums =[int(x) for x in input().split()]

# Contar o tempo
# begin_time = datetime.datetime.now()

quase_primos = 0

count = n
while count > 0:
    divide = False
    for i in range(k):
        if count % nums[i] == 0:
            divide = True
            break

    if not divide:
        quase_primos += 1

    if (count % 2 == 1):
        count -= 1
    count -= 1

print(quase_primos)

# Tempo
# print(datetime.datetime.now() - begin_time)

File: Exercicios/quase_primo.cpp
#include <iostream>
using namespace std;

int main(int argc, char const *argv[]) {
  int n, k;
  cin >> n >> k;

  int nums[k];
  for (size_t i = 0; i < k; ++i){
    cin >> nums[i];
  }

  int quase_primos = 0;

  for (size_t i = n; i > 0; i--) {
    int divide = 0;
    for (size_t j = 0; j < k; j++) {
      if(i % nums[j] == 0){
        divide = 1;
        break;
      }
    }

    if(!divide){
      quase_primos++;
    }
  }

  cout << quase_primos << endl;

  return 0;
}
